src/vxworks/mod.rs

```diff
@@ -434,33 +385,58 @@ s! {
         pub pw_shell: *mut ::c_char,
     }
 
+    // epoll.h
+     pub struct epoll_event {
+        pub events: u32,
+        pub u64: u64,
+    }
+
+    pub struct Dl_info {
+        pub dli_fname: *const ::c_char,
+        pub dli_fbase: *mut ::c_void,
+        pub dli_sname: *const ::c_char,
+        pub dli_saddr: *mut ::c_void,
+    }
+}
+
+s_no_extra_traits! {
+    // dirent.h
+    #[allow(missing_debug_implementations)]
```

Comment:
You probably need to manually implement `Debug` for these. This is quite common; you can see how other modules do it.

Reply:
I see in many files, the big size arrary is ignored so far in the debug function, for example:
        impl ::fmt::Debug for sockaddr_un {
            fn fmt(&self, f: &mut ::fmt::Formatter) -> ::fmt::Result {
                f.debug_struct("sockaddr_un")
                    .field("sun_family", &self.sun_family)
                // FIXME: .field("sun_path", &self.sun_path)                                                                         
                    .finish()
            }
        }

Should I do it in the same way? Do we have a workable solution for these big size arrary?

Reply:
IIRC there are a couple of examples that use large arrays there.

Reply:
Can you give an examaple? I can't find any of them so far.

Reply:
Sure: https://github.com/rust-lang/libc/blob/9b936e30116da305b799a74491c82ee2d2e8f678/src/unix/linux_like/linux/mod.rs#L749

You just print it as a slice `&[..]`

Reply:
Thanks.